src/main/java/tars/Storage.java

```diff
@@ -0,0 +1,116 @@
+package tars;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Handles the storing of the Tasks added to the list and retrieves information of Tasks when reopened
+ *
+ * @author csk
+ * @version 1
+ */
+public class Storage {
+    protected static File taskFile;
+
+    /**
+     * Constructor for Storage object which takes in a filePath and sets filePath field in object
+     *
+     * @param filePath
+     * @throws IOException
+     */
+    public Storage(String filePath) throws IOException {
+        this.taskFile = new File(filePath); // create a File for the given file path
+
+        if (!taskFile.exists()) {
+            taskFile.getParentFile().mkdirs();
+            taskFile.createNewFile();
+        }
+    }
+
+    /**
+     * Writes file that will be used for the Tars application based on tasks existing or changed in tasksList
+     *
+     * @param filePath
+     * @param tasks
+     * @throws IOException
+     */
+    public static void writeFile(String filePath, ArrayList<Task> tasks) throws IOException {
+        FileWriter fileWriter = new FileWriter(filePath);
+        for (Task task : tasks) {
+            fileWriter.write(task.toFileFormat() + System.lineSeparator());
+        }
+        fileWriter.close();
+    }
+
+    /**
+     * Reads file that has been passed to storage object for Tars to reuse or use for the first time
+     *
+     * @return ArrayList of tasks
+     * @throws IOException
+     */
+    public static ArrayList<Task> readFile() throws IOException {
+        Scanner s = new Scanner(taskFile); // create a Scanner using the File as the source
```

Comment:
since this is not temporary, you may want to give a more descriptive name like scanner (as compared to s) for clarity
not crucial since scanner is not used outside of this file 😛